src/vs/workbench/parts/debug/electron-browser/debugHover.ts

```diff
@@ -112,26 +112,67 @@ export class DebugHoverWidget implements editorbrowser.IContentWidget {
 			return;
 		}
 
-		// string magic to get the parents of the variable (a and b for a.b.foo)
 		const lineContent = model.getLineContent(pos.lineNumber);
-		const namesToFind = lineContent.substring(0, lineContent.indexOf('.' + hoveringOver))
-			.split('.').map(word => word.trim()).filter(word => !!word);
-		namesToFind.push(hoveringOver);
-		namesToFind[0] = namesToFind[0].substring(namesToFind[0].lastIndexOf(' ') + 1);
+		const session = this.debugService.getActiveSession();
+		let evaluatedExpression : TPromise<Expression> = undefined;
+		let matchingExpression: string = undefined;
+		let startOffset: number = 0;
+		if (session.configuration.capabilities.supportsEvaluateForHovers) {
+			let expression: RegExp = /([^()\[\]{}<>\s+\-/%~#^;=|,`!]|\->)+/g;
```

Comment:
I would suggest adding some comments indicating what this code is doing. Something like:

Line 121:
// If the debug adapter supports evaluation-based-hover, we need to try and guess what the expression under the mouse cursor is. Someday it might be nice if the language service could somehow provide this. But this code attempts to approximate the answer for a variety of languages. Some example supported expressions: myVar.prop, a.b.c.d, myVar?.prop, myVar->prop, MyClass::StaticProp, *myVar

Reg ex explanation
// Match any character *except* a set of characters which often break interesting sub-expressions

Line 124 comment:
// First find the full expression under the cursor

Line 135 comment:
// If there are non-word characters after the cursor, we want to truncate the expression then.